src/nix/parse.cc

```diff
@@ -0,0 +1,122 @@
+// based on: nix-instantiate --parse
+
+#include "command.hh"
+#include "eval.hh"
+
+using namespace nix;
+
+enum class OutputFormat {
+    ATerm,
+    XML,
+    JSON,
+};
+
+struct CmdParse : Command
+{
+    OutputFormat outputFormat = OutputFormat::ATerm;
+    std::string inputExpr;
+    Path inputFile;
+
+    CmdParse()
+    {
+        expectArgs({
+            .label = "input-file",
+            .optional = true,
+            .handler = {&inputFile},
+            .completer = completePath // libutil/args.hh
+        });
+
+        addFlag({
+            .longName = "expr",
+            //.shortName = "E", // FIXME error: cannot convert ‘<brace-enclosed initializer list>’ to ‘nix::Args::Flag&&’
+            .description = "Nix expression",
+            .labels = {"expression"},
+            .handler = {[&](std::string exprString) {
+                if (exprString.empty()) throw UsageError("--expr requires one argument");
+                inputExpr = exprString;
+                // TODO move string?
+            }}
+        });
+
+        addFlag({
+            .longName = "output-format",
+            .description = "output format",
+            .labels = {"format"},
+            .handler = {[&](std::string formatName) {
+                if (formatName.empty()) throw UsageError("--output-format requires one argument");
+                if (formatName == "json") {
+                    outputFormat = OutputFormat::JSON;
+                    return;
+                }
+                if (formatName == "aterm") {
+                    outputFormat = OutputFormat::ATerm;
+                    return;
+                }
+                throw UsageError(fmt("--output-format: format not recognized: %1%", formatName));
+            }}
+        });
+    }
+
+    std::string description() override
+    {
+        return "parse a Nix expression";
+    }
+
+    std::string doc() override
+    {
+        return
+          #include "parse.md"
+          ;
+    }
+
+    Category category() override { return catUtility; }
+
+    void run() override
+    {
+        RepairFlag repair = NoRepair;
+
+        auto store = openStore();
+
+        Strings searchPath = {"."}; // TODO better?
+
+        auto state = std::make_unique<EvalState>(searchPath, store);
+        state->repair = repair;
```

Comment:
That is the default value of `repair`. Why do you define it explicitly?

Reply:
copy-paste from nix-instantiate, consider it removed ; )